drivers/serial/uart_nrfx_uarte.c

```diff
@@ -112,6 +137,10 @@ struct uarte_async_tx {
 
 struct uarte_async_rx {
 	struct k_timer timer;
+#ifdef CONFIG_HAS_NORDIC_DMM
```

Comment:
there are several comments regarding indiscriminate usage of `dmm` APIs everywhere VS small potential savings in state variables. If you find them valuable they can stay, I guess